src/main/java/hudson/plugins/git/GitSCM.java

```diff
@@ -1038,7 +1038,7 @@ public void checkout(Run<?, ?> build, Launcher launcher, FilePath workspace, Tas
 
         BuildData previousBuildData = getBuildData(build.getPreviousBuild());   // read only
         BuildData buildData = copyBuildData(build.getPreviousBuild());
-        build.addAction(buildData);
```

Comment:
Yes - but with this change, we wait to add it as an action until after that
processing, so we're sure it's unique.